kong/plugins/oauth2/access.lua

```diff
@@ -376,16 +376,21 @@ function _M.execute(conf)
     end
   end
 
-  local token = retrieve_token(parse_access_token(conf))
+  local accessToken = parse_access_token(conf);
+  if not accessToken then
+    return responses.send_HTTP_UNAUTHORIZED({}, false, {["WWW-Authenticate"] = 'Bearer realm="service"'})
+  end
+
+  local token = retrieve_token(accessToken)
   if not token then
-    return responses.send_HTTP_FORBIDDEN("Invalid authentication credentials")
+    return responses.send_HTTP_UNAUTHORIZED({[ERROR] = "invalid_token", error_description = "The access token invalid"}, false, {["WWW-Authenticate"] = 'Bearer realm="service" error="invalid_token" error_description="The access token invalid"'})
```

Comment:
On a side note, I don't see `invalid_token` as one of the possible error types, as specified in the spec: https://tools.ietf.org/html/rfc6749#section-5.2